src/t9p_mem.c

```diff
@@ -0,0 +1,152 @@
+/**
+* ----------------------------------------------------------------------------
+ * Company    : SLAC National Accelerator Laboratory
+ * ----------------------------------------------------------------------------
+ * Description: t9p platform implementation using the POSIX API
+ * ----------------------------------------------------------------------------
+ * This file is part of 't9p'. It is subject to the license terms in the
+ * LICENSE.txt file found in the top-level directory of this distribution,
+ * and at:
+ *    https://confluence.slac.stanford.edu/display/ppareg/LICENSE.html.
+ * No part of 't9p', including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE.txt file.
+ * ----------------------------------------------------------------------------
+ **/
+#include "t9p_platform.h"
+#include "t9p_atomic.h"
+
+#ifdef __rtems__
+# include <rtems/malloc.h>
+#else
+# include <malloc.h>
+#endif
+
+#include <stdlib.h>
+#include <string.h>
+
+void*
+#if defined(__GNUC__) || defined(__clang__)
+__attribute__((malloc, alloc_size(1)))
+#endif
+t9p_malloc(size_t size)
+{
+  void* ptr = malloc(size);
+#ifndef NO_MEMTRACK
+  t9p_atomic_add_u64(&g_t9p_memtrack_ctx.total_allocd_bytes, malloc_usable_size(ptr));
```

Comment:
`malloc_usable_size` is unavailable in newlib and RTEMS. Should just replace this with `size` on RTEMS.